frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,311 @@
+    const { items } = this.props;
+
+    return _.reduce(filters, (counts, category, key) => {
+      if (key === 'name') {
+        return counts;
+      }
+      // Count
+      // Apply filters for all properties except for this property
+      const itemsByCategory = _.map(itemsFilteredByCategory, (categoryItems, categoryKey) => categoryKey === key ? items : categoryItems);
+      const filteredByName = this.filterByName(_.intersection(...itemsByCategory), filters);
+      const count = _.countBy(filteredByName, key);
+
+      counts[key] = _.reduce(category, (categoryCounts, value, elem) => {
+        categoryCounts[elem] = count[elem] || 0;
+        return categoryCounts;
+      }, {});
+
+      return counts;
+    }, {});
+  }
+
+  // getFilterCounts and filterItems are kept for convenience
+  // For any case where both filterCounts and filteredItems are updated together
+  // use getFilterItemsAndCounts
+  getFilterCounts(filters) {
+    return this.getFilterItemsAndCounts(filters).filterCounts;
+  }
+
+  filterItems(filters) {
+    return this.getFilterItemsAndCounts(filters).filteredItems;
+  }
+
+  clearFilters() {
+    const filters = _.cloneDeep(this.state.filters);
+    _.forOwn(filters, (category, key) => {
+      if (key === 'name') {
+        _.set(filters, 'name', {
+          active: false,
+          value: '',
+        });
+      } else {
+        _.forOwn(category, (cItem, elem) => {
+          _.set(filters, [key, elem, 'active'], false);
+        });
+      }
+    });
+    this.setState({filters});
+  }
+
+  onFilterChange(filterType, id, value) {
+    const filters = _.cloneDeep(this.state.filters);
+    if (filterType === 'name') {
+      const active = !!value;
+      filters[filterType] = { active, value };
+    } else {
+      filters[filterType][id].active = value;
+    }
+    this.setState({filters});
+  }
+
+  renderTiles() {
+    const items = this.state.filteredItems;
+
+    return (
+      <CatalogTileView.Category totalItems={items.length} viewAll={true}>
+        {_.map(items, item => {
+          const { obj, name, imgUrl, iconClass, provider, description } = item;
+          const uid = obj.metadata.uid;
+          const normalizedIconClass = iconClass && `icon ${normalizeIconClass(iconClass)}`;
+          const vendor = provider ? `Provided by ${provider}` : null;
+          return <CatalogTile
+            id={uid}
+            key={uid}
+            title={name}
+            iconImg={imgUrl}
+            iconClass={normalizedIconClass}
+            vendor={vendor}
+            description={description}
+          />;
+        })}
+      </CatalogTileView.Category>
+    );
+  }
+
+  render() {
+    const { filters, filterCounts, filteredItems } = this.state;
+
+    return (
+      <div className="co-catalog-page">
+        <div className="co-catalog-page__tabs">
+          <FilterSidePanel>
+            {_.map(filters, (filter, category) => {
+              if (category === 'name') {
```

Comment:
The name filter should be added prior to the `_.map` of filters to ensure it is always first.